addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -1258,8 +1258,62 @@ var BasicModel = AbstractModel.extend({
             parentID: list.id,
             viewType: list.viewType,
         };
+
+        var position = options && options.position || 'top';
+
+        // We compute the list (get) to apply the pending changes before doing our work.
+        // We use raw: true because we only need to load the first level of relation.
+        var computedList = this.get(list.id, {raw: true});
+
+        var fieldsInfoList = computedList.fieldsInfo.list;
+
+        var hasHandle = false;
+        var handleField = null;
+
+        // We decided to check if there is an handle widget from here instead of pulling it up, to make sure we get the one from the correct view,
+        // and not from the subview or anything else like that.
+        for (var field in fieldsInfoList) {
+            if(fieldsInfoList[field].widget === 'handle') {
+                if(!hasHandle) {
+                    hasHandle = true;
+                    handleField = fieldsInfoList[field].name;
+                    // if there are 2 handle fields on the same list, we take the first one we find (and that will be alphabetically on the field name...)
+                    // that being said, there might be some check somewhere else to make sure this does not happen (apparently not though).
+                    // TODO: document this behaviour.
+                    break;
+                }
+            }
+        }
+
+        // So if we have an handle widget and at least a line on the list, the currently added element must be at the bottom (or top) of the current page.
+        if (hasHandle && computedList.data.length) {
+            // Also we don't want to force the handle field if the list is not ordered by that field.
+            var isOrderedByHandle = computedList.orderedBy.length && computedList.orderedBy[0].asc === true && computedList.orderedBy[0].name === handleField;
```

Comment:
note that you can compute this without using the computedList, but with the list element (so you can compute the computedList later in the next if